plutus-metatheory/src/Main.lagda

```diff
@@ -150,11 +150,11 @@ postulate
 {-# FOREIGN GHC import Data.Functor #-}
 {-# COMPILE GHC ParseError = type PlutusCore.Error.ParserErrorBundle #-}
 
-{-# COMPILE GHC parse = parseProgram  #-}
-{-# COMPILE GHC parseU = U.parseProgram  #-}
-{-# COMPILE GHC parseTm = parseTerm  #-}
-{-# COMPILE GHC parseTy = parseType  #-}
-{-# COMPILE GHC parseTmU = U.parseTerm  #-}
+{-# COMPILE GHC parse = parseProgram . bsToText #-}
```

Comment:
Can we just change the use sites to use `Text`?

Reply:
I tried that but couldn't figure the Agda out......I will address the rest of the comments and then maybe @jmchapman can decide what to do? Changing the use sites is probably best.

Reply:
It mostly looks like it's just delegating to Haskell stuff, so it doesn't look too bad?

Reply:
I'll have a look. Agda's `String` gets compiled to `Text` by default I think so this could be a possibility to remove some more indirection. A lot of previous indirection was already removed with Marty's previous work on the parser which made the types more concrete if I recall correctly.

Reply:
Thank you very much James! If you can do without `bsToText` I can actually remove it. I'm not using it elsewhere.

Reply:
I removed the calls to bsToText and changed the interface to the plutus-core parser that the plutus-metatheory agda code is expecting to use text instead of bytestrings, I also had to update some tests - they previously used bytestrings and strings, I changed it to text and strings - I should probably just use text.

I also wondered, in general, if a switch from bytestings to text might make anything slower?

Reply:
> I also wondered, in general, if a switch from bytestings to text might make anything slower?

Not at a scale that would be noticeable, I expect.